spark/src/main/java/org/opensearch/sql/spark/execution/session/SessionManager.java

```diff
@@ -9,35 +9,31 @@
 import static org.opensearch.sql.spark.execution.session.SessionId.newSessionId;
 
 import java.util.Optional;
+import lombok.RequiredArgsConstructor;
 import org.opensearch.sql.common.setting.Settings;
 import org.opensearch.sql.spark.client.EMRServerlessClientFactory;
-import org.opensearch.sql.spark.execution.statestore.StateStore;
+import org.opensearch.sql.spark.execution.statestore.SessionStorageService;
+import org.opensearch.sql.spark.execution.statestore.StatementStorageService;
 import org.opensearch.sql.spark.utils.RealTimeProvider;
 
 /**
  * Singleton Class
  *
  * <p>todo. add Session cache and Session sweeper.
  */
+@RequiredArgsConstructor
 public class SessionManager {
-  private final StateStore stateStore;
+  private final SessionStorageService sessionStorageService;
+  private final StatementStorageService statementStorageService;
   private final EMRServerlessClientFactory emrServerlessClientFactory;
-  private Settings settings;
-
-  public SessionManager(
-      StateStore stateStore,
-      EMRServerlessClientFactory emrServerlessClientFactory,
-      Settings settings) {
-    this.stateStore = stateStore;
-    this.emrServerlessClientFactory = emrServerlessClientFactory;
-    this.settings = settings;
-  }
+  private final Settings settings;
```

Comment:
https://github.com/vamsi-amazon/sql/blob/dqs-demo/spark/src/main/java/org/opensearch/sql/spark/execution/session/SessionManager.java#L25

Can we remove dependency on settings wherever possible. We can add SessionConfigSupplier similar to the above code.

Reply:
Yes, I can address that in a separate PR.